File: app/src/main/java/me/eagzzycsl/darkroom/uitls/MyLogger.java
package me.eagzzycsl.darkroom.uitls;

import android.util.Log;

public enum MyLogger {

    accessibility("无障碍服务"),
    exception("异常"),
    freeze("冻结任务");

    public String name;

    MyLogger(String name) {
        this.name = name;
    }

    private void i(String s) {
        Log.i(this.name, s);
    }


    private String stringify(Object object) {
        String type = object == null ? "null" :
                object instanceof String ? "String" : "Object";
        String log = object == null ? "null" : object.toString();

        return type + " | " + log;
    }

    public void i(Object object) {
        if (MyConfig.DEBUG) {
            this.i(this.stringify(object));
        }
    }

    public void i(String tag, Object object) {
        if (MyConfig.DEBUG) {
            this.i(tag + "@: " + this.stringify(object));
        }
    }
}

File: app/src/main/java/me/eagzzycsl/darkroom/utils/MyLogger.kt
package me.eagzzycsl.darkroom.utils

import android.util.Log

enum class MyLogger(name: String) {

    Accessibility("无障碍服务"),
    Action("模拟操作"),
    Exception("异常"),
    Network("网络"),
    Notification("通知"),
    Freeze("冻结任务");

    fun i(s: String) {
        Log.i(this.name, s)
    }

    fun e(s: String) {
        Log.e(this.name, s);
    }

    fun i(tag: String, content: String) {
        Log.i(this.name, tag + "@" + content);
    }

    fun e(tag: String, content: String) {
        Log.i(this.name, tag + "@" + content);
    }
}
